spec/models/mixins/ci_feature_mixin_spec.rb

```diff
@@ -0,0 +1,61 @@
+describe CiFeatureMixin do
+  let(:service) { FactoryGirl.create(:service) }
+  describe "#retireable?" do
+    context "with vm" do
+      let(:vm) { FactoryGirl.create(:vm) }
+
+      it "is retireable" do
+        resource = FactoryGirl.create(:service_resource, :resource_type => "VmOrTemplate", :service_id => service.id, :resource_id => vm.id)
```

Comment:
You should be able to use the relations here rather than setting the columns directly.  `FactoryGirl.create(:service_resource, :service => service, :resource => vm)`